nixos/modules/services/mail/postal.nix

```diff
@@ -0,0 +1,530 @@
+          wantedBy = [ "postal.target" ];
+          partOf = [ "postal.target" ];
+          environment.POSTAL_CONFIG_FILE_PATH = postalConfigFile;
+        };
+
+        serviceConfigDefaults = {
+          User = cfg.user;
+          EnvironmentFile = optionals (cfg.environmentFile != null) cfg.environmentFile;
+
+          # hardening
+          BindReadOnlyPaths =
+            [
+              "/etc"
+              builtins.storeDir
+            ]
+            ++ optional config.services.mysql.enable "/run/mysqld/mysqld.sock"
+            ++ optional (
+              cfg.settings.smtp_server.tls_certificate_path != null
+            ) cfg.settings.smtp_server.tls_certificate_path
+            ++ optional (
+              cfg.settings.smtp_server.tls_private_key_path != null
+            ) cfg.settings.smtp_server.tls_private_key_path;
+          CapabilityBoundingSet = "";
+          DevicePolicy = "closed";
+          LockPersonality = true;
+          MemoryDenyWriteExecute = true;
+          NoNewPrivileges = true;
+          PrivateDevices = true;
+          PrivateTmp = true;
+          ProcSubset = "pid";
+          ProtectClock = true;
+          ProtectControlGroups = true;
+          ProtectHome = true;
+          ProtectHostname = true;
+          ProtectKernelLogs = true;
+          ProtectKernelModules = true;
+          ProtectKernelTunables = true;
+          ProtectProc = "invisible";
+          ProtectSystem = "strict";
+          RemoveIPC = true;
+          RestrictAddressFamilies = [
+            "AF_INET"
+            "AF_INET6"
+            "AF_NETLINK"
+            "AF_UNIX"
+          ];
+          RestrictNamespaces = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          RootDirectory = "/run/postal";
+          SystemCallArchitectures = "native";
+          SystemCallFilter = [
+            "@system-service"
+            "~@privileged"
+          ];
+          UMask = "0066";
+        };
+
+      in
+      {
+        postal-update = serviceDefaults // {
+          after = optional config.services.mysql.enable "mysql.service";
+          bindsTo = optional config.services.mysql.enable "mysql.service";
+          environment.POSTAL_CONFIG_FILE_PATH = postalConfigFile;
+          serviceConfig = serviceConfigDefaults // {
+            Type = "oneshot";
+            RemainAfterExit = true;
+            ExecStart = "${pkgs.postal}/bin/postal update";
+          };
+        };
+
+        postal-smtp = serviceDefaults // {
+          serviceConfig = serviceConfigDefaults // {
+            ExecStart = "${pkgs.postal}/bin/postal smtp-server";
+            AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" ];
+            CapabilityBoundingSet = [ "CAP_NET_BIND_SERVICE" ];
+          };
+        };
+
+        postal-web = serviceDefaults // {
+          serviceConfig = serviceConfigDefaults // {
+            ExecStart = "${pkgs.postal}/bin/postal web-server";
```

Comment:
```suggestion
            Type = "notify";
            ExecStart = "${lib.getExe pkgs.postal} web-server";
```

Reply:
I reverted `Type = "notify"`, I had a look at this but didn't understand well what's the goal, can you explain please?

Reply:
This is a systemd feature that allow the service to notify systemd when the service is ready not just when it's started. Puma support this

Reply:
I had to replace `eval` by `exec` in the postal executable (done in the package), this avoids to create a sub-process and puma now works with systemd-notify. I checked the other services and they don't support it, seems good to me.